datamapplot/deckgl_template.html

```diff
@@ -576,10 +587,71 @@
             fontWeight: {{font_weight}},
             lineSpacing: {{line_spacing}},
             textCollisionSizeScale: {{text_collision_size_scale}},
+            pickable: true,
           });
 
+          {% if enable_table_of_contents %}
+            const tocContainer = document.querySelector('#table-of-contents');
+            
+            {% if table_of_contents_on_click %}
+            const toc = new TableOfContents(
+                tocContainer,
+                datamap,
+                true,
+                "{{table_of_contents_button_icon}}"                
+            );
+              
+            // Set up the table of contents button behaviour.
+            toc.container.querySelectorAll('.toc-btn').forEach(button => {
+                button.addEventListener('click', function() {
+                    var label = datamap.labelData.find(l => {
+                      return l.id === this.dataset.labelId
+                    })
+                    {{table_of_contents_on_click}}
+                    // console.log(
+                    //     label.points[0].map(x=>datamap.metaData.hover_text[x])
+                    // );
+                });
+            });
+            {% else %}
+            const toc = new TableOfContents(
+                tocContainer,
+                datamap,
+                false,
+                null                
+            );
+            {% endif -%}
+            
+            function debounce(func, wait) {
+                let timeout;
+                return function executedFunction(...args) {
+                    const later = () => {
+                        clearTimeout(timeout);
+                        func(...args);
+                    };
+                    clearTimeout(timeout);
+                    timeout=setTimeout(later, wait);
+                };
+            }
+            const debouncedViewStateChange = debounce(({viewState, interactionState}) => {
+                const userIsInteracting = Object.values(interactionState).every(Boolean);
+                if (!userIsInteracting) {
+                    const visible = getVisibleTextData(viewState)
+                    if (visible) {
+                        toc.highlightElements(visible)
+                    };
+                };
+            }, 250);
```

Comment:
With the other performance improvements we can make this smaller and performance still seems pretty good to me.

```suggestion
            }, 50);
```